### DISCUSSION TITLE
How to return tsvectors with prisma?

### DISCUSSION THREAD

**Discussion by special-character** [prisma/prisma#3074] [Q&A] [answered]
I need to do full text search with prisma but after setting up my text indexes and rankings I get the error:

```
error deserializing column 19: cannot convert between the Rust type `core::option::Option<alloc::string::String>` and the Postgres type `tsvector`
```

After looking around a bit, it seems that these types are not supported fully: https://github.com/prisma/prisma/issues/1690

How should I handle reading the `tsvector`?

**Comment by special-character** [comment, marked as answer]
I actually learned if I don't put the `tsvector` column in the select statement it will run just fine. You can still use the column to compute other things as well and return them like rank and highlight but just can't actually read the `tsvector` column

**Reply by janpio** [reply]
Glad you figured that out.

Does this behavior make sense to you right now?

**Reply by special-character** [reply]
Yeah, I think this makes sense for why it doesn't work to return it from the db (and make it to prisma). I am still curious on how to handle types that prisma doesn't have out of the box but I was able to accomplish what I needed by using the tsvector on the DB side to compute a value and return a number.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
